Updated notebook with map2alm analysis

javierhndev opened the pull request:
I've been analyzing the cl spectrum on different maps (3D vs 2D, smaller resolution...). It looks good. But I'm still thinking about the last part: writing a file image and reading it back. Creating an image file from the numpy array looks fine when is read and plotted, but some information is lost (wcs?) and the alm (and cl) cannot be calculated again. 

For the AI training we will generate thousands of image files. The model then can read those images. However we need to calculate a metric (cl spectrum) to be able to quantify the performance of the AI model.

That wcs (or something else) should be stored somewhere. Need to think about this. Maybe save it as a json file?

zonca commented:
@javierhndev the standard way is to use `pixell.write_map`, this saves the wcs, then read it back with `pixell.read_map`, I recommend to use `fmt="hdf"` to use HDF5

zonca reviewed (changes requested):
I see both notebooks modified, please remove modifications on one of the notebook if you do not use it.

also, you should crop the image using pixell, so that the wcs is properly handled.
https://github.com/simonsobs/pixell_tutorials/blob/master/Pixell_map_manipulation.ipynb see the Selecting regions of the sky section.

Then you just need to save 1 map with `write_map` in HDF format. Then you can save the others in png. Then when you need to analyze the maps, you can get the `wcs` from the reference map that you saved.

zonca commented:
@javierhndev also consider that if you are doing `map2alm` in a patch of the sky instead of the full sky, you should apodize it: https://github.com/simonsobs/pixell_tutorials/blob/master/Pixell_spherical_harmonics.ipynb

javierhndev commented:
I created a new commit with the new changes. To your comments:
- When cropping the image using the standard Python slicing, the wcs is automatically adjusted (I checked and that was the expected behavior according to the tutorial).
- I'm writing the map as an HDF file and I can read it without issues.
- I'm apodizing the cropped map following the pixell tutorial

zonca commented:
can you also run `nbstripout` on the notebook to remove the outputs, then paste the executed version to gist with `gh gist create --public` (or manually) and put link here in the issue and inside the blank notebook.

zonca commented:
also please create a new notebook, I prefer to keep also the other notebooks around for reference. Please rename the noteboks with the month so they are sorted, e.g. 20205_xxxxx.ipynb

zonca commented:
I went through the notebook, it looks all good.
It is normal that the power spectrum of the center of the galaxy is larger than the entire map, being the brightest region.

So just address my 2 previous comments and we are good to merge. Thanks!

javierhndev commented:
I've created a new notebook starting wit the *202505_*. I removed the output as well with `nbstripout` and added a [link to gist](https://gist.github.com/javierhndev/b58920472e7906110585fb6cb6b92b74) with the executed version. 

I also included a requirments.txt for the virtual environment
